### PR TITLE
Add Kotlin data capturing samples

### PR DESCRIPTION
**Pull request by alextu**
This adds:
- a sample settings.gradle.kts
- sample Kotlin scripts to capture data, equivalent to the Groovy ones

Notes:
- the `gradle-quality-check-issues.gradle.kts` is quite complicated, the FindBugs part has been removed since replaced by SpotBugs. Here how it looks for:
    - [Checkstyle](https://ge.solutions-team.gradle.com/s/ngfq2kc36xe7o/custom-values)
    - [SpotBugs](https://ge.solutions-team.gradle.com/s/65xd4tnpogawg/custom-values)
    - [CodeNarc](https://ge.solutions-team.gradle.com/s/hrmovn57p4kcq/custom-values)
- only the `gradle-dependency-resolution.gradle.kts` has not been manually tested (not sure how to create such build)
- this is not an idiomatic Kotlin port, probably could be simplified here and there

### CODE REVIEW

File: .github/workflows/gradle-data-capturing-samples-verification.yml [hunk 1]
```diff
@@ -42,3 +42,10 @@ jobs:
       - name: Run Gradle build
```
Comment (jprinet) [inline .github/workflows/gradle-data-capturing-samples-verification.yml:42]: I would precise that this is the Gradle sample step

File: .github/workflows/gradle-data-capturing-samples-verification.yml [hunk 2]
```diff
@@ -42,3 +42,10 @@ jobs:
...
         working-directory: common-gradle-enterprise-gradle-configuration
         run: ./gradlew tasks
+      - name: Inject data capture Kotlin script into Gradle build
+        run: |
+          # apply sample file
+          echo "apply from: file(\"../build-data-capturing-gradle-samples/${{matrix.sample-file}}.kts\")" >> common-gradle-enterprise-gradle-configuration/build.gradle
```
Comment (jprinet) [inline .github/workflows/gradle-data-capturing-samples-verification.yml:48]: I'd rather not name the kotlin sample .`gradle.kts` and have the extension added depending on the GH action step.

File: build-data-capturing-gradle-samples/capture-git-diffs/gradle-git-diffs.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        gradle.rootProject.logger.warn("User has not set 'gistToken'. Cannot publish gist.")
+        return
+    }
+
+    val diff = execAndGetStdout("git", "diff")
+    if (!diff.isNullOrEmpty()) {
+        try {
+            val baseUrl = java.net.URL("https://api.github.com/gists")
+            val credentials = gradle.rootProject.findProperty("gistToken") as String
+            val auth = "token " + credentials
```
[the remark this answers is not in the document]
Comment (jprinet) [reply build-data-capturing-gradle-samples/capture-git-diffs/gradle-git-diffs.gradle.kts:45]: would probably make sense to align both impl

**Review by jprinet — APPROVED** [review]
I'm not a Kotlin expert, maybe another review from someone else could be interesting

File: .github/workflows/gradle-data-capturing-samples-verification.yml (hunk 2 above)
Comment (alextu) [reply .github/workflows/gradle-data-capturing-samples-verification.yml:48]: How would you organize the Kotlin files ? Currently they are sitting next to their groovy counterpart with same name, different extension (.kts)

Comment (jprinet) [reply .github/workflows/gradle-data-capturing-samples-verification.yml:48]: My impression is that a different extension is enough to differentiate the content. I don't think that a separate hierarchy would really help.

Comment (alextu) [reply .github/workflows/gradle-data-capturing-samples-verification.yml:48]: ah then I'm not sure to understand your initial comment, could you clarify ?

File: build-data-capturing-gradle-samples/capture-dependency-resolution/gradle-dependency-resolution.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,29 @@
+
+/**
+ * This Gradle script captures any dependency configurations that are resolved early to build the task graph,
+ * and adds these as custom values.
+ * This should be applied to the root project:
+ * <code> apply from: file('gradle-dependency-resolution.gradle.kts') </code>
+ */
+
+
+project.extensions.configure<com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension>() {
```
Comment (runningcode) [inline build-data-capturing-gradle-samples/capture-dependency-resolution/gradle-dependency-resolution.gradle.kts:10]: is it possible to use imports here?

File: .github/workflows/gradle-data-capturing-samples-verification.yml (hunk 2 above)
Comment (jprinet) [reply .github/workflows/gradle-data-capturing-samples-verification.yml:48]: I would name the samples `foo.kts` and not `foo.gradle.kts`

Comment (alextu) [reply .github/workflows/gradle-data-capturing-samples-verification.yml:48]: Ah, it looks like the current convention is to name Gradle Kotlin scripts `*.gradle.kts`

Comment (jprinet) [reply .github/workflows/gradle-data-capturing-samples-verification.yml:48]: my bad then, you can dismiss this comment

File: .github/workflows/gradle-data-capturing-samples-verification.yml (hunk 1 above)
Comment (alextu) [reply .github/workflows/gradle-data-capturing-samples-verification.yml]: done

File: build-data-capturing-gradle-samples/capture-dependency-resolution/gradle-dependency-resolution.gradle.kts (hunk 4 above)
Comment (alextu) [reply build-data-capturing-gradle-samples/capture-dependency-resolution/gradle-dependency-resolution.gradle.kts]: done

File: build-data-capturing-gradle-samples/capture-git-diffs/gradle-git-diffs.gradle.kts (hunk 3 above)
[the remark this answers is not in the document]
Comment (alextu) [reply build-data-capturing-gradle-samples/capture-git-diffs/gradle-git-diffs.gradle.kts:46]: I will create a seperate PR for it

**Pull request closed without merging**